mediacloud/mediawords/util/html.py

```diff
@@ -81,3 +88,107 @@ def __get_meta_refresh_url_from_tag(inner_tag: str, inner_base_url=None) -> Opti
             return url
 
     return None
+
+
+def _sententize_block_level_tags(s: str) -> str:
+    """Add a double newline after each block level tag and a newline before the end of each block level tag.
+
+    Arguments:
+    s - html string
+
+    Returns:
+    string with tags replaced.
+
+    """
+    _BLOCK_LEVEL_ELEMENT_TAGS = \
+        ('title h1 h2 h3 h4 h5 h6 p div dl dt dd ol ul li dir menu address'
+         ' blockquote center div hr ins noscript pre').split()
+    _TAG_LIST = '|'.join(_BLOCK_LEVEL_ELEMENT_TAGS)
+    _BLOCK_LEVEL_START_TAG_RE = r'(<(' + _TAG_LIST + ')(>|\s))'
+    _BLOCK_LEVEL_END_TAG_RE = r'(</(' + _TAG_LIST + ')>)'
+
+    s = re.sub(_BLOCK_LEVEL_START_TAG_RE, "\n\n\\1", s, flags=re.S | re.I)
+    s = re.sub(_BLOCK_LEVEL_END_TAG_RE, ".\\1\n\n", s, flags=re.S | re.I)
+
+    # get rid of repeat periods
+    s = re.sub(r'\.(\s*\.)+', '.', s, flags=re.S)
+    s = re.sub(r'^(\s*\.\s)+', '', s, flags=re.S)
+
+    return s
+
+
+def html_strip(s: str, include_title: bool=False) -> str:
+    """Strip the html tags, html comments, any any text within TITLE, SCRIPT, APPLET, OBJECT, and STYLE tags.
+
+    Derived from code by powerman from: http://www.perlmonks.org/?node_id=161281.
+
+    Arguments:
+    s - html to strip
+    include_title - if true, the title text in the returned text
+    """
+    s = str(decode_object_from_bytes_if_needed(s))
+
+    # help the sentence parse understand headers as individual sentences
+    s = _sententize_block_level_tags(s)
+
+    # Remove soft hyphen (&shy or 0xAD) character from text
+    # (some news websites hyphenate their stories using this character so that the browser can lay it out more nicely)
+    s = re.sub('\xAD', '', s, flags=re.S)
+
+    soup = BeautifulSoup(s, 'lxml')
+
+    remove_tags = 'script applet object style'.split()
+    if not include_title:
+        remove_tags.append('title')
+
+    for tag in soup(remove_tags):
+        tag.decompose()
+
+    text = soup.get_text(' ', strip=True)
+
+    return text.strip()
+
+
+def html_title(html: str, fallback: str, trim_to_length: int=0) -> Optional[str]:
+    """Parse the content for tags that might indicate the story's title.
```

Comment:
1)

```python
html = decode_object_from_bytes_if_needed(html)
fallback = decode_object_from_bytes_if_needed(fallback)
```

2) Maybe add an extra `int()` around `trim_to_length` for a good measure.